File: webapp-net/SmartTarget/Models/SmartTargetPromotion.cs
using Sdl.Web.Common.Configuration;
using Sdl.Web.Common.Models;
using System;
using System.Collections.Generic;

namespace Sdl.Web.Modules.SmartTarget.Models
{
    public class SmartTargetPromotion : EntityModel
    {
        private const string _xpmMarkupFormat = "<!-- Start Promotion: {{ \"PromotionID\": \"{0}\", \"RegionID\" : \"{1}\"}} -->";

        public string RegionName { get; set; }

        public string PromotionId { get; set; }

        public string Title { get; set; }

        public string Slogan { get; set; }

        public List<SmartTargetItem> Items { get; set; }

        public override string GetXpmMarkup(Localization localization)
        {
            return string.Format(_xpmMarkupFormat, PromotionId, RegionName);
        }
    }
}

File: webapp-java/dxa-module-smarttarget-abstract/src/main/java/com/sdl/dxa/modules/smarttarget/model/entity/SmartTargetPromotion.java
package com.sdl.dxa.modules.smarttarget.model.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.common.collect.Collections2;
import com.sdl.dxa.caching.NeverCached;
import com.sdl.webapp.common.api.localization.Localization;
import com.sdl.webapp.common.api.model.EntityModel;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Collection;
import java.util.List;
import java.util.Map;

import static org.springframework.util.CollectionUtils.isEmpty;

/**
 * <p>SmartTargetPromotion class.</p>
 */
@EqualsAndHashCode(callSuper = true)
@Data
@NeverCached(qualifier = "SmartTargetPromotion")
public class SmartTargetPromotion extends AbstractEntityModel {

    @JsonProperty("Title")
    private String title;

    @JsonProperty("Slogan")
    private String slogan;

    @JsonProperty("Items")
    private List<SmartTargetItem> items;

    /**
     * {@inheritDoc}
     */
    @Override
    public String getXpmMarkup(Localization localization) {
        Map<String, Object> xpmMetadata = getXpmMetadata();
        if (isEmpty(xpmMetadata)) {
            return "";
        }

        return String.format("<!-- Start Promotion: { \"PromotionID\": \"%s\", \"RegionID\" : \"%s\" } -->",
                xpmMetadata.get("PromotionID"), xpmMetadata.get("RegionID"));
    }

    @JsonIgnore
    public Collection<EntityModel> getEntityModels() {
        return Collections2.transform(this.items, SmartTargetItem::getEntity);
    }

}
